### DISCUSSION TITLE
(a current) Libreoffice in SpiralLinux

### DISCUSSION THREAD

**Discussion by AlphaElwedritsch** [SpiralLinux/SpiralLinux-project#349] [Q&A] [answered]
Hi there,

To get a current LibreOffice in SpiralLinux, does it make sense to integrate this ppa, or rather use manual *.deb files from the LibreOffice download page?
https://launchpad.net/~libreoffice/+archive/ubuntu/ppa

Or flatpak installation?

Or is there another way to get it onto the system natively (without going through flatpak)?

What do you think it's best? Best Practice?

Thanks

**Comment by geckolinux** [comment, marked as answer]
Hi there,

I'm not a fan of Flatpak either as a principal means of app distribution, but it's definitely the easiest way in this case. The latest LibreOffice is available out-of-the-box in SpiralLinux via Flathub in Gnome Software or KDE Discover.

I definitely would not recommend the Ubuntu PPA.

Another option is via the pre-installed Debian Backports repository. It should be possible via Synaptic to force installation of the version from Backports, but it doesn't work, so unfortunately it requires the command line:
`sudo apt install -t bookworm-backports libreoffice`

**Reply by AlphaElwedritsch** [reply]
Hi there,
thanks
in the meatime I researched more for myself

I agree with you in all points.

And just in that moment you answered here I did it that way:
`sudo apt upgrade libreoffice -t bookworm-backports`
what's at the end the same you suggested. Only that I upgraded...

**Reply by Anonymo** [reply]
Would adding the fasttrack repo do a similar thing?

https://fasttrack.debian.net/

**Reply by geckolinux** [reply]
@Anonymo FastTrack is enabled by default for SpiralLinux, but it mainly serves to provide packages for VirtualBox. The normal Debian Backports repo is the only official Debian source for newer LibreOffice packages.

**Comment by AlphaElwedritsch** [comment]
changed to Flatpak version.
7.6 has a lot of improvements that I don't want to miss. The 7.5 is too old for me.

**Comment by AlphaElwedritsch** [comment]
I think, regardless of whether this is a good or bad development, that in the future Flatpak, Snap, Appimage will play an increasingly larger role in package management. at least as far as the big third-party software packages are concerned.
This is being pushed quite heavily by the big distros.
what do you think? or am I seeing this completely wrong?

**Reply by geckolinux** [reply]
Yeah, in fact most of the big distros are already relying on them and/or strongly recommending them to their users. I suspect that Debian and Arch will probably remain more conservative in that regard, although both of them do allow the user to install and use those package formats if desired.

**Comment by AlphaElwedritsch** [comment]
bookworm-backports has updated from 7.5 to newest 7.6.4
can change back now from flatpak to native :)

**Reply by geckolinux** [reply]
Thanks, can I ask how you installed it? Were you able to install it via Synaptic?

**Reply by AlphaElwedritsch** [reply]
I deinstalled the flatpak: `flatpak uninstall LibreOffice`
and installed it with: `sudo apt install -t bookworm-backports libreoffice libreoffice-I10n-de libreoffice-gtk3`
without libreoffice-gtk3 it looks ugly

I didn't try to install it via synaptics, did directly the manual way

**Comment by pertsavk** [comment]
A bit late, but regarding the question if using that particular PPA would be a good idea - No. No way. There is the telltale keyword "ubuntu" in the PPA address and that means it's a big NO. Don't. Do. It.
Probably all you who answered before me already knew this but some day there might be someone who doesn't so I'll just leave this warning here. Years ago I didn't know and karma hit me badly. Never just grab an Ubuntu PPA and use it in Debian, unless you know exactly what you are doing and are ready to face the possible consequences. And if you know what you are doing you probably don't do it.

**Comment by AlphaElwedritsch** [comment]
Hi there @geckolinux 
If upgraded LO from backports `sudo apt upgrade -t bookworm-backports libreoffice libreoffice-l10n-de libreoffice-gtk3`
Will it be automatically updated if there is a new version in backports, or di I have to do it manually?
Mean, is the backports repo fullfill integrated in the 'sudo apt upgrade' mechanism?
thanks

**Reply by geckolinux** [reply]
Hi there, I'm honestly not sure. Please manually compare the latest available version that Synaptic shows you and compare it to the version that gets installed after following the [recommended update process](https://github.com/SpiralLinux/SpiralLinux-project/wiki#updates-within-a-debian-stable-release-lifecycle).

This information about package priority from Backports might be helpful:
https://wiki.debian.org/Backports#Using_backports

Also feel free to ask this question on a Debian forum or mailing list. You're running a legitimate customized Debian Stable system with all packages unmodified coming directly from Debian package repos, and there are no SpiralLinux package repos or package modifications, so questions and answers about this topic for Debian are applicable to your installation too.

**Reply by AlphaElwedritsch** [reply]
No problem
The forum here is always my first port of call ;)
